lib/elixir/lib/exception.ex

```diff
@@ -1079,6 +1079,27 @@ defmodule BadArityError do
 end
 
 defmodule UndefinedFunctionError do
+  @moduledoc """
+  An exception raised when a function is invoked that is not defined.
+
+  The following fields of this exception are public and can be used freely:
+
+    * `:module` (`t:module/0`) - the module name
+    * `:function` (`t:atom/0`) - the function name
+    * `:arity` (`t:non_neg_integer/0`) - the arity of the function
```

Comment:
Should we go ahead and add a `@type t` to document the types? Or is it awkward with dialyzer because we have private fields too? (maybe we can make other fields opaque, ...)

Reply:
I avoided it on purpose because yeah, seems more trouble then it's worth. Better if folks just use the `Exception.t/0` type IMO, until someone comes screaming after us for this 😀